src/common/keyboard_layout.cpp

```diff
@@ -1,276 +1,304 @@
-        keyCodes.push_back(VK_MENU);
-        keyCodes.push_back(VK_LMENU);
-        keyCodes.push_back(VK_RMENU);
-        keyCodes.push_back(VK_SHIFT);
-        keyCodes.push_back(VK_LSHIFT);
-        keyCodes.push_back(VK_RSHIFT);
-        // Add character keys
-        for (auto& it : unicodeKeys)
-        {
-            // If it was not renamed with a special name
-            if (it.second == keyboardLayoutMap[it.first])
-            {
-                keyCodes.push_back(it.first);
-            }
-        }
-        // Add all other special keys
-        for (int i = 1; i < 256; i++)
-        {
-            // If it is not already been added (i.e. it was either a modifier or had a unicode representation)
-            if (std::find(keyCodes.begin(), keyCodes.end(), i) == keyCodes.end())
-            {
-                // If it is any other key but it is not named as VK #
-                auto it = unknownKeys.find(i);
-                if (it == unknownKeys.end())
-                {
-                    keyCodes.push_back(i);
-                }
-                else if (unknownKeys[i] != keyboardLayoutMap[i])
-                {
-                    keyCodes.push_back(i);
-                }
-            }
-        }
-        // Add unknown keys
-        for (auto& it : unknownKeys)
-        {
-            // If it was not renamed with a special name
-            if (it.second == keyboardLayoutMap[it.first])
-            {
-                keyCodes.push_back(it.first);
-            }
-        }
-        keyCodeList = keyCodes;
-        isKeyCodeListGenerated = true;
-    }
-    else
-    {
-        keyCodes = keyCodeList;
-    }
-
-    // If it is a key list for the shortcut control then we add a "None" key at the start
-    if (isShortcut)
-    {
-        keyCodes.insert(keyCodes.begin(), 0);
-    }
-
-    return keyCodes;
-}
-
-Windows::Foundation::Collections::IVector<Windows::Foundation::IInspectable> LayoutMap::GetKeyNameList(const bool isShortcut)
-{
-    std::unique_lock<std::mutex> lock(keyboardLayoutMap_mutex);
-    UpdateLayout();
-    lock.unlock();
-    Windows::Foundation::Collections::IVector<Windows::Foundation::IInspectable> keyNames = single_threaded_vector<Windows::Foundation::IInspectable>();
-    std::vector<DWORD> keyCodes = GetKeyCodeList(isShortcut);
-    lock.lock();
-    // If it is a key list for the shortcut control then we add a "None" key at the start
-    if (isShortcut)
-    {
-        keyNames.Append(winrt::box_value(L"None"));
-        for (int i = 1; i < keyCodes.size(); i++)
-        {
-            keyNames.Append(winrt::box_value(keyboardLayoutMap[keyCodes[i]].c_str()));
-        }
-    }
-    else
-    {
-        for (int i = 0; i < keyCodes.size(); i++)
-        {
-            keyNames.Append(winrt::box_value(keyboardLayoutMap[keyCodes[i]].c_str()));
-        }
-    }
-
-    return keyNames;
-}
+#include "pch.h"
+#include "keyboard_layout_impl.h"
+
+#pragma comment(lib, "user32.lib")
```

Comment:
Does adding this have any effect on using the header on the cpp side as well?

Reply:
I removed it and set the `interop` to link the default libs instead, which includes `user32.lib`